packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {APIGatewayProxyEvent, APIGatewayProxyResult} from "aws-lambda"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {injectLambdaContext} from "@aws-lambda-powertools/logger/middleware"
+import {DynamoDBClient, PutItemCommand} from "@aws-sdk/client-dynamodb"
+import {marshall} from "@aws-sdk/util-dynamodb"
+import middy from "@middy/core"
+import inputOutputLogger from "@middy/input-output-logger"
+import errorHandler from "@nhs/fhir-middy-error-handler"
+import {v4 as uuidv4} from "uuid"
+
+const logger = new Logger({serviceName: "updatePrescriptionStatus"})
+const client = new DynamoDBClient({region: "eu-west-2"})
+const tableName = process.env.TABLE_NAME
+
+const MISSING_FIELDS_RESPONSE: APIGatewayProxyResult = {
+  statusCode: 400,
+  body: JSON.stringify({error: "Missing required fields"}),
+  headers: {
+    "Content-Type": "application/fhir+json",
+    "Cache-Control": "no-cache"
+  }
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    // Parse request body
```

Comment:
This try catch is a bit far reaching, it wraps most of the lambdas logic but only catches for a single error type. This could hide errors with dynamo or other parts of the logic and makes logging, monitoring and debugging harder, as well as potentially returning incorrect information back to the client. The catch statement itself is potentially unsafe and could catch other errors, or fail to catch your intended error.

Reply:
Validation will be implemented in another ticket.